File: server/Models/VaultKeep.cs
using System.ComponentModel.DataAnnotations;

namespace keepr.Models;

public class VaultKeep : RepoItem<int>
{
  public int KeepId { get; set; }
  public int VaultId { get; set; }
  [MaxLength(255)] public string CreatorId { get; set; }
  // public Profile Creator { get; set; }

}

File: client/src/models/VaultKeep.js
import { APIItem } from "./APIItem.js";
import { Profile } from "./Profile.js";


export class VaultKeep extends APIItem{
constructor(data){
  super(data)
  this.keepId = data.keepId
  this.vaultId = data.vaultId
      this.vaultKeepId = data.vaultKeepId
      this.name = data.name
      this.description = data.description
      this.img = data.img
      this.views = data.views
      this.kept = data.kept
      this.creatorId = data.creatorId
      this.creator = new Profile(data.creator)
}
}